edit_course/operations/configure.py

```diff
@@ -497,22 +498,20 @@ def get_target_category(category, course=None, course_url=None):
 
     else:
 
-        course_slug_begin = course_url.find('/', course_url.find('://') + 1) + 1
+        parsed_url = urlparse(course_url)
+        service_hostname = urlparse(settings.BASE_URL).hostname
 
-        if settings.BASE_URL:
-            # Can correctly identify slug-beginning with BASE_URLs
-            # even if they end in '/'
-            course_slug_begin = course_url.find('/', len(settings.BASE_URL) - 1) + 1
+        if parsed_url.hostname != service_hostname:
+            return None, _("Course URL '{}' doesn't match service's hostname '{}'".format(course_url, service_hostname))
+
+        course_slug = parsed_url.path.split('/')[1]
+        instance_slug = parsed_url.path.split('/')[2]
```

Comment:
You can do this with single split (python is unable to do optimisation here).
`course_slug, instance_slug = parsed_url.path.split('/')[:2]` (will fail if there is no `/` in the path).